repos/fdbt-site/src/pages/api/csvUpload.ts

```diff
@@ -218,32 +268,116 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
+            if (ticket !== undefined && matchingJsonMetaData !== undefined) {
+                const fareZoneNames = getNamesOfFareZones(ticket);
+
+                if (!fareZoneNames) {
+                    return;
+                }
+
+                const errors: ErrorInfo[] = [];
+
+                const thereIsANameMismatch = thereIsAFareStageNameMismatch(fareTriangleData, fareZoneNames);
+
+                // check to see if the the number of fare stages does not match
+                if (fareTriangleData.fareStages.length !== fareZoneNames.length) {
+                    errors.push(stageCountMismatchError);
+                } else if (thereIsANameMismatch) {
+                    errors.push(nameMismatchError);
+                }
+
+                // check to see if we had errors
+                if (errors.length > 0) {
+                    setCsvUploadAttributeAndRedirect(req, res, errors, fields.poundsOrPence as string);
+                    return;
+                }
+
+                // clear the errors from the session
+                updateSessionAttribute(req, CSV_UPLOAD_ATTRIBUTE, { errors: [] });
+
+                // update the fare zones on the matching json
+                if ('fareZones' in ticket) {
+                    ticket.fareZones = getFareZones(
+                        fareTriangleData,
+                        ticket.fareZones.reduce<MatchingFareZones>((matchingFareZones, currentFareZone) => {
+                            matchingFareZones[currentFareZone.name] = currentFareZone;
+                            return matchingFareZones;
+                        }, {}),
+                    );
+                } else {
+                    ticket.inboundFareZones = getFareZones(
+                        fareTriangleData,
+                        ticket.inboundFareZones.reduce<MatchingFareZones>((matchingFareZones, currentFareZone) => {
+                            matchingFareZones[currentFareZone.name] = currentFareZone;
+                            return matchingFareZones;
+                        }, {}),
+                    );
+
+                    ticket.outboundFareZones = getFareZones(
+                        fareTriangleData,
+                        ticket.outboundFareZones.reduce<MatchingFareZones>((matchingFareZones, currentFareZone) => {
+                            matchingFareZones[currentFareZone.name] = currentFareZone;
+                            return matchingFareZones;
+                        }, {}),
+                    );
+                }
+
+                // put the now updated matching json into s3
+                // overriding the existing object
+                await putUserDataInProductsBucketWithFilePath(ticket, matchingJsonMetaData.matchingJsonLink);
+
+                redirectTo(
+                    res,
+                    `/products/productDetails?productId=${matchingJsonMetaData?.productId}&serviceId=${matchingJsonMetaData?.serviceId}`,
+                );
```

Comment:
I think we need a return after here, as it's a redirect.

Reply:
Do we? it didn't have a `return` after the redirect before?